### PR TITLE
Refactor/settings

### PR DESCRIPTION
**Pull request by mandaiy**
yet another pull req for the setting activity.

breaking changes:
- add RxJava/RxAndroid (good bye AsyncTask!)
- layout is changed (two-pane to single-pane with dialog)
- remove "Try to Scan" button in DeviceSelectFragment (formerly called)
  - when the dialog opened, bluetooth device scanning automatically starts (how do you think?)

there are still many hard-coding strings, incoherent naming.
please comment

### CODE REVIEW

File: app/src/main/java/com/pileproject/drive/setting/app/ProgramListFragment.java [hunk 1]
```diff
@@ -80,17 +92,25 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
 
+
         return v;
     }
 
     @Override
     public void onActivityCreated(Bundle savedInstanceState) {
         super.onActivityCreated(savedInstanceState);
 
+        final ProgramListFragment thiz = this;
```
Comment (mandaiy) [inline app/src/main/java/com/pileproject/drive/setting/app/ProgramListFragment.java]: seems ugly. any workaround?

**Comment by mandaiy** [comment]
I'd like to discuss
- the behavior of the scanning process (that automatically starts when the user opens the dialog)
- the names of DeviceSelect\* (tentatively BluetoothDevice*)
  with you.
  Any opinions?

File: app/src/main/java/com/pileproject/drive/setting/app/ProgramListFragment.java (hunk 1 above)
Comment (tiwanari) [inline app/src/main/java/com/pileproject/drive/setting/app/ProgramListFragment.java]: DriveApplication.java can provide a context but it is not good way for this, I think. View v has its context and you can access it with getContext method.

**Comment by tiwanari** [comment]
Question, during search, can users choose a device or go to other page (I mean "can the process stop correctly?")
BluetoothDevice seems good. Is there any problem ?

File: app/src/main/java/com/pileproject/drive/setting/app/ProgramListFragment.java (hunk 1 above)
Comment (mandaiy) [inline app/src/main/java/com/pileproject/drive/setting/app/ProgramListFragment.java]: It can't because AlertDialogFragment.Builuder needs the parent fragment instance rather than a context in this case.
I think another resort is maintaining 'this' as a field variable, but it gets uglier I think...

**Comment by mandaiy** [comment]
> Question, during search, can users choose a device or go to other page (I mean "can the process stop correctly?")

Of course they can.

> BluetoothDevice seems good. Is there any problem ?

there's no problem, but I need your approval ;)
maybe this naming scheme is suitable, since there could be 'WifiDevice' or something.
or it could be 'BluetoothMachine'. which is preferable?

File: app/src/main/java/com/pileproject/drive/setting/app/ProgramListFragment.java (hunk 1 above)
Comment (tiwanari) [inline app/src/main/java/com/pileproject/drive/setting/app/ProgramListFragment.java]: Oh, I was confused... I have no good idea now;( what should we do in normal (?) way? Something like this: http://stackoverflow.com/questions/22777928/button-click-inside-fragment-to-open-up-new-fragment ?

**Comment by tiwanari** [comment]
Ok, I also think we should discuss about using "machine" or "device," and in my opinion, machine is better to keep code consistence. 
I have one more question, what should users do when they try to search again?

**Comment by mandaiy** [comment]
> Ok, I also think we should discuss about using "machine" or "device," and in my opinion, machine is better to keep code consistence. 

👍

> I have one more question, what should users do when they try to search again?

Now this app continually scans bluetooth devices (in other words, users do not need to restart it)
To be precise, the scanning restarts when it ends

https://github.com/myusak/drive/blob/30cffb6bb15e0f8add521d8e3645301af4db78c3/app/src/main/java/com/pileproject/drive/setting/machine/BluetoothDeviceSelectFragment.java#L323-L347

When the broadcast receiver receives 'BluetoothAdapter.DISCOVERY_FINISHED', 'onNext' is called.
Then, it calls BluetoothAdapter#startDiscovery() again.

**Comment by tiwanari** [comment]
Perfect :)

File: app/src/main/java/com/pileproject/drive/setting/app/ProgramListFragment.java (hunk 1 above)
Comment (mandaiy) [inline app/src/main/java/com/pileproject/drive/setting/app/ProgramListFragment.java]: 'ProgramListFragment.this' worked. 
really embarrassed... (and that's why i hate this language)

**Comment by mandaiy** [comment]
I hope you accept my pull request.

**Comment by tiwanari** [comment]
thanks for a notification:) give me time for check.

File: app/src/androidTest/AndroidManifest.xml [hunk 2]
```diff
@@ -1,4 +1,3 @@
-<?xml version="1.0" encoding="utf-8"?>
```
Comment (tiwanari) [inline app/src/androidTest/AndroidManifest.xml:1]: Why did you remove this?

File: app/src/main/java/com/pileproject/drive/programming/visual/activity/ProgrammingActivityBase.java [hunk 3]
```diff
@@ -118,43 +129,37 @@ private void findViews() {
...
-        mSpaceManager.deleteAllBlocks(); // delete existing blocks
-        if (isSample) { // load a sample program
-            mSpaceManager.loadSampleProgram(selectedText);
-        } else { // load a user program
-            mSpaceManager.loadUserProgram(selectedText);
-        }
+    private void moveToTitleActivity() {
+        mSpaceManager.saveExecutionProgram();
+
+        // TODO: go to title activity
```
Comment (tiwanari) [inline app/src/main/java/com/pileproject/drive/programming/visual/activity/ProgrammingActivityBase.java:155]: Do you plan to keep this as TODO in this PR?

File: app/src/main/res/layout/fragment_bluetooth_machine_select.xml [hunk 4]
```diff
@@ -0,0 +1,108 @@
+<?xml version="1.0" encoding="utf-8"?><!--
```
Comment (tiwanari) [inline app/src/main/res/layout/fragment_bluetooth_machine_select.xml]: add a break before "<!--"

**Comment by tiwanari** [comment]
And this PR seems to conflict with the previous PR. Please fix it.

**Comment by tiwanari** [comment]
Though there are some kinds of stuff to be fixed, totally LGTM. APPROVED.

File: app/src/main/java/com/pileproject/drive/programming/visual/activity/ProgrammingActivityBase.java (hunk 3 above)
Comment (mandaiy) [inline app/src/main/java/com/pileproject/drive/programming/visual/activity/ProgrammingActivityBase.java:155]: The app does nothing when this button is pressed even before my pull request.
I didn't come up with a neat way to jump to the title, since the intent differs according
to the concrete title activity.
Can you give me the formal way?

**Comment by mandaiy** [comment]
What are 'some kinds of stuff to be fixed'?
It sounds like the PR contains more issues other than you mentioned.
I'd like to fix these points.

File: app/src/main/java/com/pileproject/drive/programming/visual/activity/ProgrammingActivityBase.java (hunk 3 above)
Comment (tiwanari) [inline app/src/main/java/com/pileproject/drive/programming/visual/activity/ProgrammingActivityBase.java:155]: i understood. so, i think it is good to keep almoslll blank however

Comment (mandaiy) [inline app/src/main/java/com/pileproject/drive/programming/visual/activity/ProgrammingActivityBase.java:155]: Please fix it on your excellent PR :)

**Comment by tiwanari** [comment]
I'm sorry for confusing you... I mean "there are kinds of some stuff to be fixed (as I mentioned above)". So, all are fixed and let's close this PR. I love it:)
